cy_ioctl.go

```diff
@@ -158,45 +158,35 @@ func (d *Device) doIoctl(kind uint32, iface whd.IoctlInterface, cmd whd.SDPCMCom
 	if err != nil {
 		return err
 	}
-	start := time.Now()
-	const ioctlTimeout = 50 * time.Millisecond
-	const maxRetries = 3
-	for retries := 0; time.Since(start) < ioctlTimeout || retries < maxRetries; retries++ {
-		payloadOffset, plen, header, err := d.sdpcmPoll(d.buf[:])
-		Debug("doIoctl:sdpcmPoll conclude payloadoffset=", int(payloadOffset), "plen=", int(plen), "header=", header.String(), err)
-		// The "wrong payload type" appears to happen during startup. See sdpcmProcessRxPacket
-		if err != nil && !errors.Is(err, err9WrongPayloadType) {
-			continue
-		}
-		payload := d.buf[payloadOffset : payloadOffset+plen]
-		switch header {
-		case whd.CONTROL_HEADER:
-			n := copy(buf[:], payload)
-			if uint32(n) != plen {
-				return errors.New("not enough space on ioctl buffer for control header copy")
+
+	timeout := time.NewTicker(50 * time.Millisecond)
+
+	for {
+		select {
+		case <-timeout.C:
+			return errDoIoctlTimeout
+		case ctrlHdr := <-d.ctrlCh:
```

Comment:
Why are we switching over to goroutines for dealing with this? It seemed to be solved OK via normal function calls, or was there something to gain?